url/nodeurl.go

```diff
@@ -0,0 +1,118 @@
+package url
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type searchParam struct {
+	name  string
+	value []string
```

Comment:
I've been going though the code and added some comments, and at some point I realise the most important question is why do you need a slice here? I leave the other comments in as most of them are still relevant, however this one needs to be addressed first.

Reply:
Good point, I'll simply this and remove the use of an array. This is a relic from before when I didn't maintain ordering of parameters.